app/helpers/document_type_helper.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module DocumentTypeHelper
+  def t_doctype_key_exists?(i18n_key)
+    I18n.exists?(i18n_key) || I18n.exists?(default_key(i18n_key))
+  end
+
+  def t_doctype(i18n_key)
+    match = I18n.t!(I18n.exists?(i18n_key) ? i18n_key : default_key(i18n_key))
+    match.is_a?(String) ? match : match.stringify_keys
```

Comment:
Could you help we understand what we need this for?

Reply:
It's used here: `TagField.new(tag_config.merge(t_doctype(tag_i18n_data)))`

We're trying to merge tag metadata:

```
    - id: primary_publishing_organisation
      type: single_tag
      document_type: organisation
```

...with i18n info:

```
        primary_publishing_organisation:
          label: Primary organisation
          hint: The organisation responsible for publishing and maintaining this content.
```

Rather than manually retrieving the i18n value for both `label` and `hint`, this code pulls the i18n value for `primary_publishing_organisation` as a whole, so that we can merge _all_ the i18n values into the tag metadata.

But by default I18n will return hashes as symbol/value pairs instead of string/value pairs, which is different to the hashes we retrieve from parsing the YAML in our `self.all` method. So I use `stringify_keys` on the hash.

Reply:
I think I noticed this after I commented 😄. However, since the main goal is still to get this stuff out of the models, could we look at getting the views to do this instead?

Reply:
`tag_field.label` is called in _tags.html, _single_tag_input.html and _multi_tag_input.html. If I use `t` in the views, I get errors like this (for both label and hint):

`translation missing: en.document_types.8b5139bd.fields.tag_id.label`

...which I can solve by adding an arbitrary default label (doesn't feel like the right approach!):

```ruby
t("document_types.#{@edition.document_type.id}.fields.#{tag_field.id}.label", default: "Tag label missing")
```

We could come to the same conclusion as we came to for the document type labels - i.e. to keep each tag `label`/`hint` in config/document_types.yml. Or we can keep this model code which combines the metadata and the I18n keys.

Reply:
Ah, that's annoying. I think this is happening because the tests are creating random IDs, which won't be possible for much longer (at least, not for content fields). We could reasonably change them to use a known ID, since the total set of fields isn't meant to be dynamic. Does that help?

Reply:
I can look into it tomorrow morning 🤔 also need to fix my build after all the refactoring. Hope to wrap this up soon!

Reply:
I have now done this 👍 thanks for the suggestion.